repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/glossary/TermPreProcessor.java

```diff
@@ -120,12 +147,97 @@ private void processUpdateTerm(AtlasEntity entity, AtlasVertex vertex) throws At
         RequestContext.get().endMetricRecord(metricRecorder);
     }
 
+    private boolean checkEntityTermAssociation(String termQName) throws AtlasBaseException{
+        List<AtlasEntityHeader> entityHeader;
+
+        try {
+             entityHeader = fetchEntityHeadersByTermQualifiedName(0,1,termQName);
+        } catch (AtlasBaseException e) {
+            throw e;
+        }
+
+        Boolean hasEntityAssociation = entityHeader != null ? true : false;
+
+        return hasEntityAssociation;
+    }
+
+    private Map<String, Object> getMap(String key, Object value) {
+        Map<String, Object> map = new HashMap<>();
+        map.put(key, value);
+        return map;
+    }
+
+    private List<AtlasEntityHeader> fetchEntityHeadersByTermQualifiedName(int from, int size, String termQName) throws AtlasBaseException{
+        IndexSearchParams indexSearchParams = new IndexSearchParams();
+        Map<String, Object> dsl = getMap("from", from);
+        dsl.put("size", size);
+        dsl.put("query", getMap("term", getMap("__meanings", getMap("value",termQName))));
+        indexSearchParams.setDsl(dsl);
+        AtlasSearchResult searchResult = null;
+        try {
+            searchResult = discovery.directIndexSearch(indexSearchParams);
+        } catch (AtlasBaseException e) {
+            throw e;
+        }
+        List<AtlasEntityHeader> entityHeaders = searchResult.getEntities();
+        return  entityHeaders;
+    }
+
+    public void updateMeaningsNamesInEntities(String updatedTermName, String termQName, String termGuid) throws AtlasBaseException {
+        int from = 0;
+        while (true) {
+            List<AtlasEntityHeader> entityHeaders = fetchEntityHeadersByTermQualifiedName(from, ELASTICSEARCH_PAGINATION_OFFSET, termQName);
+            if (entityHeaders == null)
+                break;
+            for (AtlasEntityHeader entityHeader : entityHeaders) {
+                   String updatedMeaningsText = entityHeader
+                           .getMeanings()
+                           .stream()
+                           .map(x -> x.getTermGuid().equals(termGuid) ? updatedTermName : x.getDisplayText())
+                           .collect(Collectors.joining(","));
+                AtlasGraphUtilsV2.setEncodedProperty(AtlasGraphUtilsV2.findByGuid(entityHeader.getGuid()), MEANINGS_TEXT_PROPERTY_KEY, updatedMeaningsText);
+            }
+            from += ELASTICSEARCH_PAGINATION_OFFSET;
+
+            if (entityHeaders.size() < ELASTICSEARCH_PAGINATION_OFFSET)
+                break;
+        }
+
+    }
+
+    public void createAndQueueTask(String taskType, String updatedTermName, String termQName, AtlasVertex termVertex) {
+        String termGuid = GraphHelper.getGuid(termVertex);
+        String currentUser = RequestContext.getCurrentUser();
+        Map<String, Object> taskParams = MeaningsTask.toParameters(updatedTermName, termQName, termGuid, ELASTICSEARCH_PAGINATION_OFFSET);
```

Comment:
This is public static constant & can be from anywhere directly, no need to store it on task vertex